Flaky metrics test makes CI fails

Latest master [build](https://console.cloud.google.com/cloud-build/builds/ea36a93b-50d0-4458-9f8a-d1a0039df515?project=stargateio) pointed again the flaky tests. We should add awaitability to these tests and add a small amount of wait time.

```
Step #4 - "dse-6.8": INFO  [main] 2021-07-27 09:01:19,912 ExternalStorage.java:288 - Storage cluster (version 6.8) that was requested by [engine:junit-jupiter]/[class:io.stargate.it.http.graphql.cqlfirst.ApolloTest] has been removed.
Step #4 - "dse-6.8": [INFO] 
Step #4 - "dse-6.8": [INFO] Results:
Step #4 - "dse-6.8": [INFO] 
Step #4 - "dse-6.8": [ERROR] Failures: 
Step #4 - "dse-6.8": [ERROR]   MetricsTest.restApiHttpRequestMetrics:82 
Step #4 - "dse-6.8": Expecting any element of:
Step #4 - "dse-6.8":   <["http_server_requests_seconds{keyspaceName="unknown",method="GET",module="health-checker",status="200",tagme="UNKNOWN",uri="/metrics",quantile="0.95",} 0.855638016",
Step #4 - "dse-6.8":     "http_server_requests_seconds{keyspaceName="unknown",method="GET",module="health-checker",status="200",tagme="UNKNOWN",uri="/metrics",quantile="0.99",} 0.855638016",
Step #4 - "dse-6.8":     "http_server_requests_seconds_count{keyspaceName="unknown",method="GET",module="health-checker",status="200",tagme="UNKNOWN",uri="/metrics",} 8.0",
Step #4 - "dse-6.8":     "http_server_requests_seconds_sum{keyspaceName="unknown",method="GET",module="health-checker",status="200",tagme="UNKNOWN",uri="/metrics",} 4.596390149",
Step #4 - "dse-6.8":     "http_server_requests_seconds{keyspaceName="unknown",method="POST",module="authapi",status="400",tagme="test-value",uri="/v1/auth",quantile="0.95",} 0.08388608",
Step #4 - "dse-6.8":     "http_server_requests_seconds{keyspaceName="unknown",method="POST",module="authapi",status="400",tagme="test-value",uri="/v1/auth",quantile="0.99",} 0.08388608",
Step #4 - "dse-6.8":     "http_server_requests_seconds_count{keyspaceName="unknown",method="POST",module="authapi",status="400",tagme="test-value",uri="/v1/auth",} 1.0",
Step #4 - "dse-6.8":     "http_server_requests_seconds_sum{keyspaceName="unknown",method="POST",module="authapi",status="400",tagme="test-value",uri="/v1/auth",} 0.084741833",
Step #4 - "dse-6.8":     "http_server_requests_seconds{keyspaceName="unknown",method="GET",module="graphqlapi",status="401",tagme="test-value",uri="/graphql",quantile="0.95",} 0.0524288",
Step #4 - "dse-6.8":     "http_server_requests_seconds{keyspaceName="unknown",method="GET",module="graphqlapi",status="401",tagme="test-value",uri="/graphql",quantile="0.99",} 0.0524288",
Step #4 - "dse-6.8":     "http_server_requests_seconds_count{keyspaceName="unknown",method="GET",module="graphqlapi",status="401",tagme="test-value",uri="/graphql",} 1.0",
Step #4 - "dse-6.8":     "http_server_requests_seconds_sum{keyspaceName="unknown",method="GET",module="graphqlapi",status="401",tagme="test-value",uri="/graphql",} 0.052870841",
Step #4 - "dse-6.8":     "http_server_requests_seconds{keyspaceName="someKeyspace",method="POST",module="graphqlapi",status="401",tagme="test-value",uri="/graphql/{keyspaceName}",quantile="0.95",} 0.0016384",
Step #4 - "dse-6.8":     "http_server_requests_seconds{keyspaceName="someKeyspace",method="POST",module="graphqlapi",status="401",tagme="test-value",uri="/graphql/{keyspaceName}",quantile="0.99",} 0.0016384",
Step #4 - "dse-6.8":     "http_server_requests_seconds_count{keyspaceName="someKeyspace",method="POST",module="graphqlapi",status="401",tagme="test-value",uri="/graphql/{keyspaceName}",} 1.0",
Step #4 - "dse-6.8":     "http_server_requests_seconds_sum{keyspaceName="someKeyspace",method="POST",module="graphqlapi",status="401",tagme="test-value",uri="/graphql/{keyspaceName}",} 0.001697407",
Step #4 - "dse-6.8":     "http_server_requests_seconds_max{keyspaceName="unknown",method="GET",module="health-checker",status="200",tagme="UNKNOWN",uri="/metrics",} 0.862193079",
Step #4 - "dse-6.8":     "http_server_requests_seconds_max{keyspaceName="unknown",method="POST",module="authapi",status="400",tagme="test-value",uri="/v1/auth",} 0.084741833",
Step #4 - "dse-6.8":     "http_server_requests_seconds_max{keyspaceName="unknown",method="GET",module="graphqlapi",status="401",tagme="test-value",uri="/graphql",} 0.052870841",
Step #4 - "dse-6.8":     "http_server_requests_seconds_max{keyspaceName="someKeyspace",method="POST",module="graphqlapi",status="401",tagme="test-value",uri="/graphql/{keyspaceName}",} 0.001697407"]>
Step #4 - "dse-6.8": to satisfy the given assertions requirements but none did:
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds{keyspaceName="unknown",method="GET",module="health-checker",status="200",tagme="UNKNOWN",uri="/metrics",quantile="0.95",} 0.855638016"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds{keyspaceName="unknown",method="GET",module="health-checker",status="200",tagme="UNKNOWN",uri="/metrics",quantile="0.95",} 0.855638016">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"module="restapi""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds{keyspaceName="unknown",method="GET",module="health-checker",status="200",tagme="UNKNOWN",uri="/metrics",quantile="0.99",} 0.855638016"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds{keyspaceName="unknown",method="GET",module="health-checker",status="200",tagme="UNKNOWN",uri="/metrics",quantile="0.99",} 0.855638016">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"module="restapi""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds_count{keyspaceName="unknown",method="GET",module="health-checker",status="200",tagme="UNKNOWN",uri="/metrics",} 8.0"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds_count{keyspaceName="unknown",method="GET",module="health-checker",status="200",tagme="UNKNOWN",uri="/metrics",} 8.0">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"module="restapi""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds_sum{keyspaceName="unknown",method="GET",module="health-checker",status="200",tagme="UNKNOWN",uri="/metrics",} 4.596390149"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds_sum{keyspaceName="unknown",method="GET",module="health-checker",status="200",tagme="UNKNOWN",uri="/metrics",} 4.596390149">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"module="restapi""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds{keyspaceName="unknown",method="POST",module="authapi",status="400",tagme="test-value",uri="/v1/auth",quantile="0.95",} 0.08388608"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds{keyspaceName="unknown",method="POST",module="authapi",status="400",tagme="test-value",uri="/v1/auth",quantile="0.95",} 0.08388608">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"method="GET""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds{keyspaceName="unknown",method="POST",module="authapi",status="400",tagme="test-value",uri="/v1/auth",quantile="0.99",} 0.08388608"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds{keyspaceName="unknown",method="POST",module="authapi",status="400",tagme="test-value",uri="/v1/auth",quantile="0.99",} 0.08388608">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"method="GET""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds_count{keyspaceName="unknown",method="POST",module="authapi",status="400",tagme="test-value",uri="/v1/auth",} 1.0"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds_count{keyspaceName="unknown",method="POST",module="authapi",status="400",tagme="test-value",uri="/v1/auth",} 1.0">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"method="GET""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds_sum{keyspaceName="unknown",method="POST",module="authapi",status="400",tagme="test-value",uri="/v1/auth",} 0.084741833"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds_sum{keyspaceName="unknown",method="POST",module="authapi",status="400",tagme="test-value",uri="/v1/auth",} 0.084741833">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"method="GET""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds{keyspaceName="unknown",method="GET",module="graphqlapi",status="401",tagme="test-value",uri="/graphql",quantile="0.95",} 0.0524288"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds{keyspaceName="unknown",method="GET",module="graphqlapi",status="401",tagme="test-value",uri="/graphql",quantile="0.95",} 0.0524288">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"module="restapi""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds{keyspaceName="unknown",method="GET",module="graphqlapi",status="401",tagme="test-value",uri="/graphql",quantile="0.99",} 0.0524288"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds{keyspaceName="unknown",method="GET",module="graphqlapi",status="401",tagme="test-value",uri="/graphql",quantile="0.99",} 0.0524288">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"module="restapi""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds_count{keyspaceName="unknown",method="GET",module="graphqlapi",status="401",tagme="test-value",uri="/graphql",} 1.0"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds_count{keyspaceName="unknown",method="GET",module="graphqlapi",status="401",tagme="test-value",uri="/graphql",} 1.0">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"module="restapi""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds_sum{keyspaceName="unknown",method="GET",module="graphqlapi",status="401",tagme="test-value",uri="/graphql",} 0.052870841"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds_sum{keyspaceName="unknown",method="GET",module="graphqlapi",status="401",tagme="test-value",uri="/graphql",} 0.052870841">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"module="restapi""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds{keyspaceName="someKeyspace",method="POST",module="graphqlapi",status="401",tagme="test-value",uri="/graphql/{keyspaceName}",quantile="0.95",} 0.0016384"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds{keyspaceName="someKeyspace",method="POST",module="graphqlapi",status="401",tagme="test-value",uri="/graphql/{keyspaceName}",quantile="0.95",} 0.0016384">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"method="GET""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds{keyspaceName="someKeyspace",method="POST",module="graphqlapi",status="401",tagme="test-value",uri="/graphql/{keyspaceName}",quantile="0.99",} 0.0016384"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds{keyspaceName="someKeyspace",method="POST",module="graphqlapi",status="401",tagme="test-value",uri="/graphql/{keyspaceName}",quantile="0.99",} 0.0016384">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"method="GET""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds_count{keyspaceName="someKeyspace",method="POST",module="graphqlapi",status="401",tagme="test-value",uri="/graphql/{keyspaceName}",} 1.0"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds_count{keyspaceName="someKeyspace",method="POST",module="graphqlapi",status="401",tagme="test-value",uri="/graphql/{keyspaceName}",} 1.0">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"method="GET""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds_sum{keyspaceName="someKeyspace",method="POST",module="graphqlapi",status="401",tagme="test-value",uri="/graphql/{keyspaceName}",} 0.001697407"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds_sum{keyspaceName="someKeyspace",method="POST",module="graphqlapi",status="401",tagme="test-value",uri="/graphql/{keyspaceName}",} 0.001697407">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"method="GET""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds_max{keyspaceName="unknown",method="GET",module="health-checker",status="200",tagme="UNKNOWN",uri="/metrics",} 0.862193079"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds_max{keyspaceName="unknown",method="GET",module="health-checker",status="200",tagme="UNKNOWN",uri="/metrics",} 0.862193079">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"module="restapi""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds_max{keyspaceName="unknown",method="POST",module="authapi",status="400",tagme="test-value",uri="/v1/auth",} 0.084741833"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds_max{keyspaceName="unknown",method="POST",module="authapi",status="400",tagme="test-value",uri="/v1/auth",} 0.084741833">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"method="GET""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds_max{keyspaceName="unknown",method="GET",module="graphqlapi",status="401",tagme="test-value",uri="/graphql",} 0.052870841"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds_max{keyspaceName="unknown",method="GET",module="graphqlapi",status="401",tagme="test-value",uri="/graphql",} 0.052870841">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"module="restapi""> 
Step #4 - "dse-6.8": 
Step #4 - "dse-6.8":   <"http_server_requests_seconds_max{keyspaceName="someKeyspace",method="POST",module="graphqlapi",status="401",tagme="test-value",uri="/graphql/{keyspaceName}",} 0.001697407"> error: 
Step #4 - "dse-6.8": Expecting:
Step #4 - "dse-6.8":  <"http_server_requests_seconds_max{keyspaceName="someKeyspace",method="POST",module="graphqlapi",status="401",tagme="test-value",uri="/graphql/{keyspaceName}",} 0.001697407">
Step #4 - "dse-6.8": to contain:
Step #4 - "dse-6.8":  <"method="GET""> 
Step #4 - "dse-6.8": [INFO] 
Step #4 - "dse-6.8": [ERROR] Tests run: 902, Failures: 1, Errors: 0, Skipped: 10
```

--- a/testing/src/main/java/io/stargate/it/http/MetricsTest.java
+++ b/testing/src/main/java/io/stargate/it/http/MetricsTest.java
@@ -17,6 +17,7 @@
 package io.stargate.it.http;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
 
 import io.stargate.it.BaseOsgiIntegrationTest;
 import io.stargate.it.storage.StargateConnectionInfo;
@@ -25,6 +26,7 @@ import io.stargate.it.storage.StargateSpec;
 import io.stargate.testing.TestingServicesActivator;
 import io.stargate.testing.metrics.TagMeHttpMetricsTagProvider;
 import java.io.IOException;
+import java.time.Duration;
 import java.util.Arrays;
 import java.util.List;
 import java.util.stream.Collectors;
@@ -71,32 +73,39 @@ public class MetricsTest extends BaseOsgiIntegrationTest {
             .build();
 
     int status = execute(client, request);
-    String result = RestUtils.get("", String.format("%s:8084/metrics", host), HttpStatus.SC_OK);
 
-    List<String> lines =
-        Arrays.stream(result.split(System.getProperty("line.separator")))
-            .filter(line -> line.startsWith("http_server_requests"))
-            .collect(Collectors.toList());
+    await()
+        .atMost(Duration.ofSeconds(5))
+        .untilAsserted(
+            () -> {
+              String result =
+                  RestUtils.get("", String.format("%s:8084/metrics", host), HttpStatus.SC_OK);
 
-    assertThat(lines)
-        .anySatisfy(
-            metric ->
-                assertThat(metric)
-                    .contains("method=\"GET\"")
-                    .contains("module=\"restapi\"")
-                    .contains("uri=\"/v1/keyspaces\"")
-                    .contains(String.format("status=\"%d\"", status))
-                    .contains(TagMeHttpMetricsTagProvider.TAG_ME_KEY + "=\"test-value\"")
-                    .contains("quantile=\"0.95\""))
-        .anySatisfy(
-            metric ->
-                assertThat(metric)
-                    .contains("method=\"GET\"")
-                    .contains("module=\"restapi\"")
-                    .contains("uri=\"/v1/keyspaces\"")
-                    .contains(String.format("status=\"%d\"", status))
-                    .contains(TagMeHttpMetricsTagProvider.TAG_ME_KEY + "=\"test-value\"")
-                    .contains("quantile=\"0.99\""));
+              List<String> lines =
+                  Arrays.stream(result.split(System.getProperty("line.separator")))
+                      .filter(line -> line.startsWith("http_server_requests"))
+                      .collect(Collectors.toList());
+
+              assertThat(lines)
+                  .anySatisfy(
+                      metric ->
+                          assertThat(metric)
+                              .contains("method=\"GET\"")
+                              .contains("module=\"restapi\"")
+                              .contains("uri=\"/v1/keyspaces\"")
+                              .contains(String.format("status=\"%d\"", status))
+                              .contains(TagMeHttpMetricsTagProvider.TAG_ME_KEY + "=\"test-value\"")
+                              .contains("quantile=\"0.95\""))
+                  .anySatisfy(
+                      metric ->
+                          assertThat(metric)
+                              .contains("method=\"GET\"")
+                              .contains("module=\"restapi\"")
+                              .contains("uri=\"/v1/keyspaces\"")
+                              .contains(String.format("status=\"%d\"", status))
+                              .contains(TagMeHttpMetricsTagProvider.TAG_ME_KEY + "=\"test-value\"")
+                              .contains("quantile=\"0.99\""));
+            });
   }
 
   @Test
@@ -112,32 +121,39 @@ public class MetricsTest extends BaseOsgiIntegrationTest {
             .build();
 
     int status = execute(client, request);
-    String result = RestUtils.get("", String.format("%s:8084/metrics", host), HttpStatus.SC_OK);
 
-    List<String> lines =
-        Arrays.stream(result.split(System.getProperty("line.separator")))
-            .filter(line -> line.startsWith("http_server_requests"))
-            .collect(Collectors.toList());
+    await()
+        .atMost(Duration.ofSeconds(5))
+        .untilAsserted(
+            () -> {
+              String result =
+                  RestUtils.get("", String.format("%s:8084/metrics", host), HttpStatus.SC_OK);
 
-    assertThat(lines)
-        .anySatisfy(
-            metric ->
-                assertThat(metric)
-                    .contains("method=\"GET\"")
-                    .contains("module=\"graphqlapi\"")
-                    .contains("uri=\"/graphql\"")
-                    .contains(String.format("status=\"%d\"", status))
-                    .contains(TagMeHttpMetricsTagProvider.TAG_ME_KEY + "=\"test-value\"")
-                    .contains("quantile=\"0.95\""))
-        .anySatisfy(
-            metric ->
-                assertThat(metric)
-                    .contains("method=\"GET\"")
-                    .contains("module=\"graphqlapi\"")
-                    .contains("uri=\"/graphql\"")
-                    .contains(String.format("status=\"%d\"", status))
-                    .contains(TagMeHttpMetricsTagProvider.TAG_ME_KEY + "=\"test-value\"")
-                    .contains("quantile=\"0.99\""));
+              List<String> lines =
+                  Arrays.stream(result.split(System.getProperty("line.separator")))
+                      .filter(line -> line.startsWith("http_server_requests"))
+                      .collect(Collectors.toList());
+
+              assertThat(lines)
+                  .anySatisfy(
+                      metric ->
+                          assertThat(metric)
+                              .contains("method=\"GET\"")
+                              .contains("module=\"graphqlapi\"")
+                              .contains("uri=\"/graphql\"")
+                              .contains(String.format("status=\"%d\"", status))
+                              .contains(TagMeHttpMetricsTagProvider.TAG_ME_KEY + "=\"test-value\"")
+                              .contains("quantile=\"0.95\""))
+                  .anySatisfy(
+                      metric ->
+                          assertThat(metric)
+                              .contains("method=\"GET\"")
+                              .contains("module=\"graphqlapi\"")
+                              .contains("uri=\"/graphql\"")
+                              .contains(String.format("status=\"%d\"", status))
+                              .contains(TagMeHttpMetricsTagProvider.TAG_ME_KEY + "=\"test-value\"")
+                              .contains("quantile=\"0.99\""));
+            });
   }
 
   @Test
@@ -153,34 +169,41 @@ public class MetricsTest extends BaseOsgiIntegrationTest {
             .build();
 
     int status = execute(client, request);
-    String result = RestUtils.get("", String.format("%s:8084/metrics", host), HttpStatus.SC_OK);
 
-    List<String> lines =
-        Arrays.stream(result.split(System.getProperty("line.separator")))
-            .filter(line -> line.startsWith("http_server_requests"))
-            .collect(Collectors.toList());
+    await()
+        .atMost(Duration.ofSeconds(5))
+        .untilAsserted(
+            () -> {
+              String result =
+                  RestUtils.get("", String.format("%s:8084/metrics", host), HttpStatus.SC_OK);
 
-    assertThat(lines)
-        .anySatisfy(
-            metric ->
-                assertThat(metric)
-                    .contains("method=\"POST\"")
-                    .contains("module=\"graphqlapi\"")
-                    .contains("uri=\"/graphql/{keyspaceName}\"")
-                    .contains(String.format("status=\"%d\"", status))
-                    .contains(TagMeHttpMetricsTagProvider.TAG_ME_KEY + "=\"test-value\"")
-                    .contains("quantile=\"0.95\"")
-                    .contains("keyspaceName=\"someKeyspace\""))
-        .anySatisfy(
-            metric ->
-                assertThat(metric)
-                    .contains("method=\"POST\"")
-                    .contains("module=\"graphqlapi\"")
-                    .contains("uri=\"/graphql/{keyspaceName}\"")
-                    .contains(String.format("status=\"%d\"", status))
-                    .contains(TagMeHttpMetricsTagProvider.TAG_ME_KEY + "=\"test-value\"")
-                    .contains("quantile=\"0.99\"")
-                    .contains("keyspaceName=\"someKeyspace\""));
+              List<String> lines =
+                  Arrays.stream(result.split(System.getProperty("line.separator")))
+                      .filter(line -> line.startsWith("http_server_requests"))
+                      .collect(Collectors.toList());
+
+              assertThat(lines)
+                  .anySatisfy(
+                      metric ->
+                          assertThat(metric)
+                              .contains("method=\"POST\"")
+                              .contains("module=\"graphqlapi\"")
+                              .contains("uri=\"/graphql/{keyspaceName}\"")
+                              .contains(String.format("status=\"%d\"", status))
+                              .contains(TagMeHttpMetricsTagProvider.TAG_ME_KEY + "=\"test-value\"")
+                              .contains("quantile=\"0.95\"")
+                              .contains("keyspaceName=\"someKeyspace\""))
+                  .anySatisfy(
+                      metric ->
+                          assertThat(metric)
+                              .contains("method=\"POST\"")
+                              .contains("module=\"graphqlapi\"")
+                              .contains("uri=\"/graphql/{keyspaceName}\"")
+                              .contains(String.format("status=\"%d\"", status))
+                              .contains(TagMeHttpMetricsTagProvider.TAG_ME_KEY + "=\"test-value\"")
+                              .contains("quantile=\"0.99\"")
+                              .contains("keyspaceName=\"someKeyspace\""));
+            });
   }
 
   @Test
@@ -196,32 +219,39 @@ public class MetricsTest extends BaseOsgiIntegrationTest {
             .build();
 
     int status = execute(client, request);
-    String result = RestUtils.get("", String.format("%s:8084/metrics", host), HttpStatus.SC_OK);
 
-    List<String> lines =
-        Arrays.stream(result.split(System.getProperty("line.separator")))
-            .filter(line -> line.startsWith("http_server_requests"))
-            .collect(Collectors.toList());
+    await()
+        .atMost(Duration.ofSeconds(5))
+        .untilAsserted(
+            () -> {
+              String result =
+                  RestUtils.get("", String.format("%s:8084/metrics", host), HttpStatus.SC_OK);
 
-    assertThat(lines)
-        .anySatisfy(
-            metric ->
-                assertThat(metric)
-                    .contains("method=\"POST\"")
-                    .contains("module=\"authapi\"")
-                    .contains("uri=\"/v1/auth\"")
-                    .contains(String.format("status=\"%d\"", status))
-                    .contains(TagMeHttpMetricsTagProvider.TAG_ME_KEY + "=\"test-value\"")
-                    .contains("quantile=\"0.95\""))
-        .anySatisfy(
-            metric ->
-                assertThat(metric)
-                    .contains("method=\"POST\"")
-                    .contains("module=\"authapi\"")
-                    .contains("uri=\"/v1/auth\"")
-                    .contains(String.format("status=\"%d\"", status))
-                    .contains(TagMeHttpMetricsTagProvider.TAG_ME_KEY + "=\"test-value\"")
-                    .contains("quantile=\"0.99\""));
+              List<String> lines =
+                  Arrays.stream(result.split(System.getProperty("line.separator")))
+                      .filter(line -> line.startsWith("http_server_requests"))
+                      .collect(Collectors.toList());
+
+              assertThat(lines)
+                  .anySatisfy(
+                      metric ->
+                          assertThat(metric)
+                              .contains("method=\"POST\"")
+                              .contains("module=\"authapi\"")
+                              .contains("uri=\"/v1/auth\"")
+                              .contains(String.format("status=\"%d\"", status))
+                              .contains(TagMeHttpMetricsTagProvider.TAG_ME_KEY + "=\"test-value\"")
+                              .contains("quantile=\"0.95\""))
+                  .anySatisfy(
+                      metric ->
+                          assertThat(metric)
+                              .contains("method=\"POST\"")
+                              .contains("module=\"authapi\"")
+                              .contains("uri=\"/v1/auth\"")
+                              .contains(String.format("status=\"%d\"", status))
+                              .contains(TagMeHttpMetricsTagProvider.TAG_ME_KEY + "=\"test-value\"")
+                              .contains("quantile=\"0.99\""));
+            });
   }
 
   @ParameterizedTest

